pkg/generate/route.go

```diff
@@ -17,23 +17,25 @@ func DefaultRoute(cr *regopapi.ImageRegistry, p *parameters.Globals) (Template,
 			Kind:       "Route",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      p.DefaultRoute.Name,
+			Name:      cr.ObjectMeta.Name + "-default-route",
 			Namespace: p.Deployment.Namespace,
 		},
 		Spec: routeapi.RouteSpec{
 			To: routeapi.RouteTargetReference{
 				Kind: "Service",
-				Name: p.Deployment.Name,
+				Name: cr.ObjectMeta.Name,
```

Comment:
we expect this service to a have a fixed name, right?  (currently docker-registry)

Reply:
Yep, but only for legacy deployments in the "default" namespace. no ?

Reply:
I would expect it to be a fixed name for new installs too (though possibly a new name, like "image-registry")?

Reply:
I don't want to worry that someone manages to create the CR with some new name before the operator creates the CR, and now we have to deal with a cluster that has a random name for the registry service/etc.  Is there a good reason *not* to hardcode the name of these things, given, as you note, that we're only allowing one instance of them anyway?

Reply:
> Is there a good reason not to hardcode the name of these things, given, as you note, that we're only allowing one instance of them anyway?

I wanted to remove hardcode the name from everywhere. I talked with @dmage about tests. This should help us in testing. After this commit we have only one global collection - images, but if you use random generated images with no blob reuse this should not be a problem.

But the service can be called the same everywhere. I will hardcode this name again.

Reply:
@bparees to solve this problem we can hardcode names for the CR (default/docker-registry and openshift-image-registry/image-registry). In this case modifications for adoption will be minimal. Old installations will still have `docker-registry.default.svc`, the new one will have `image-registry.openshift-image-registry.svc`.

The pattern $NAMESPACE-$CRNAME-resource will also allow us to run tests for the operator in parallel.